airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py

```diff
@@ -291,8 +292,8 @@ def path(self, next_page_token: Mapping[str, Any] = None, **kwargs: Any) -> str:
     transformer = TypeTransformer(TransformConfig.CustomSchemaNormalization | TransformConfig.DefaultSchemaNormalization)
 
     @default_backoff_handler(max_tries=5, factor=15)
-    def _send_http_request(self, method: str, url: str, json: dict = None, stream: bool = False):
-        headers = self.authenticator.get_auth_header()
+    def _send_http_request(self, method: str, url: str, json: dict = None, headers: dict = None, stream: bool = False):
+        headers = self.authenticator.get_auth_header() if not headers else headers | self.authenticator.get_auth_header()
```

Comment:
can you clarify why we added this logic?

Reply:
we need to add headers with `locator` in pagination for big BULK results. https://github.com/airbytehq/airbyte/pull/28781/files#diff-0572edeedd561901f8110c0f88b48efde87d1ed91247c0eb189c7229da12090dR561-R562